pkg/api/helpers.go

```diff
@@ -432,6 +432,14 @@ const (
 	// PreferAvoidPodsAnnotationKey represents the key of preferAvoidPods data (json serialized)
 	// in the Annotations of a Node.
 	PreferAvoidPodsAnnotationKey string = "scheduler.alpha.kubernetes.io/preferAvoidPods"
+
+	// AppArmorPodAnnotationKey represents the key of an AppArmor profile applied to all containers of a
+	// pod.
+	AppArmorPodAnnotationKey string = "apparmor.security.alpha.kubernetes.io/pod"
+
+	// AppArmorContainerAnnotationKeyPrefix represents the key of an AppArmor profile applied to one
+	// container of a pod.
+	AppArmorContainerAnnotationKeyPrefix string = "apparmor.security.alpha.kubernetes.io/container/"
```

Comment:
I see the scheme in the proposal (e.g. `apparmor.security.alpha.kubernetes.io/container/nginx`) but this is currently not a valid annotation key. To accommodate this scheme we would need to expand what we accept as valid label/annotation keys.

https://github.com/kubernetes/kubernetes/blob/master/pkg/util/validation/validation.go#L38